src/Board.cpp

```diff
@@ -0,0 +1,116 @@
+#include "Board.h"
+
+void Board::setBoard(const std::string& board) {
+	m_piece.resize(8); // resize the 2D vector
+	for (auto& row : m_piece) row.resize(8); // resize each row
+
+	// loop through the board string and create the corresponding piece objects
+	for (int i = 0; i < board.size(); ++i) {
+		int x = i % 8; // calculate x and y indices for the spot
+		int y = i / 8;
+		m_piece[y][x] = Factory<Piece>::create(board[i]);
+	}
+	setKingsPositions();
+}
+
+void Board::setKingsPositions() {
+	for (int i = 0; i < BoardSize; i++)
+		for (int j = 0; j < BoardSize; j++) 
+			if (m_piece[i][j] && typeid(*m_piece[i][j]) == typeid(King)) {
+				m_piece[i][j]->isWhite() ? setKingPos({ i + 1, j + 1 }, true) :
+					setKingPos({ i + 1, j + 1 }, false);
+			}
+}
+
+int Board::getResponse(const std::string& input) {
+	// parse the input string
+	Position src = { input[0] - 'a', input[1] - '1' };
+	Position dst = { input[2] - 'a', input[3] - '1' };
+
+	// board checks
+
+	// check if the src is empty
+	if (!m_piece[src.x][src.y]) return 11;
```

Comment:
Instead of returning hardcoded numbers, it would be more readable to use an enum